### DISCUSSION TITLE
Renovate ignoreUnstable to not upgrade any existing unstable versions?

### DISCUSSION THREAD

**Discussion by japj** [renovatebot/renovate#12358] [Request Help] [answered]
Hello,

I am trying to have Renovate not upgrade any packages (package.json) that are unstable, only stable packages should be upgraded by Renovate.

I had hoped that ignoreUnstable would be the right setting for this, but it seems to actually upgrade unstable packages (to other/newer unstable versions).

Is there a way to complete "stop" this unstable->unstable upgrade mechanism?

Thanks in advance!

**Comment by viceice** [comment]
Please provide more information about the manager datasource and versioning used. also the specific package.

Normally renovate won't suggest unstable versions by default, so it's probably a config issue or a bad package.

**Reply by japj** [reply]
Sorry, to not be clear enough.
This is with an npm package (hence the package.json reference)

We have a custom feed in AzureDevOps Server with unstable/preview packages that have the same major/minor/patch, but different unstable "part" based on git commit hash (this is generated by nerdbank.gitversioning).

e.g. 1.2.3-gccb5fc9b6f and 1.2.3-g18d1c8c8d3

In this case, there are several packages that are unstable, and we would prefer to manually update these, while still allowing automatic upgrades of all other packages (only upgrade the packages that have stable versioning specified).

**Reply by viceice** [reply]
please provide a concrete sample, as i still don't understand what happens and what should happen.

**Reply by rarkins** [reply]
If there's too many packages to put in `packageNames` or `packagePatterns` then you could use a package rule with [matchCurrentVersion](https://docs.renovatebot.com/configuration-options/#matchcurrentversion) and then set `"enabled": false`.

**Comment by japj** [comment, marked as answer]
I could indeed solve this by using `matchCurrentVersion` in combination with the regex `/.*-.*/` (which will match any preview version in npm)
